api/v1beta1/disruption_types.go

```diff
@@ -39,10 +39,12 @@ import (
 type DisruptionSpec struct {
 	// +kubebuilder:validation:Required
 	Count *intstr.IntOrString `json:"count"` // number of pods to target in either integer form or percent form appended with a %
-	// +kubebuilder:validation:Required
-	Selector labels.Set `json:"selector"`         // label selector
-	DryRun   bool       `json:"dryRun,omitempty"` // enable dry-run mode
-	OnInit   bool       `json:"onInit,omitempty"` // enable disruption on init
+	// +nullable
+	Selector labels.Set `json:"selector,omitempty"` // label selector
+	// +nullable
+	AdvancedSelector []metav1.LabelSelectorRequirement `json:"advancedSelector,omitempty"` // advanced label selector
```

Comment:
ping for @nathan-tournant to think about if we can represent the relationship between `Selector` and `AdvancedSelector` as a ddmark